backends/kv/reader.go

```diff
@@ -102,13 +102,47 @@ func (kv *Backend) Read(request *frames.ReadRequest) (frames.FrameIterator, erro
 		switch typedError := err.(type) {
 		case v3ioerrors.ErrorWithStatusCode:
 			if typedError.StatusCode() == http.StatusNotFound {
-				return nil, errors.New(
-					fmt.Sprintf("Failed to find a schema for table \"/%s/%s\"; "+
-						"use the `execute` 'infer' command to infer the schema and generate a schema file.",
-						request.Proto.Session.Container, request.Proto.Table))
+				if !request.Proto.InferSchema {
+					return nil, errors.New(
+						fmt.Sprintf("Failed to find a schema for table \"/%s/%s\"; "+
+							"use the `execute` 'infer' command to infer the schema and generate a schema file.",
+							request.Proto.Session.Container, request.Proto.Table))
+				}
+
+				iterInfer, err := v3ioutils.NewAsyncItemsCursor(
+					container, &input, kv.numWorkers, request.Proto.ShardingKeys, kv.logger, 10, partitions,
+					request.Proto.SortKeyRangeStart, request.Proto.SortKeyRangeEnd)
```

Comment:
use the config you added "maxRecordsInfer" instead of 10

Reply:
are you sure? i think i passed 10...